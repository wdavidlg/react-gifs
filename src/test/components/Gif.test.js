import { shallow } from "enzyme"
import Gif from '../../components/Gif'
import React from 'react';


describe('pruebas en <Gif />', () => {
    
    const title = 'Un titulo';
    const url = 'https://localhost/imagen.jpg';
    let wrapper = shallow(<Gif gif={{url, title}}/>);
    
    beforeEach(() => {
        wrapper = shallow(<Gif gif={{url, title}}/>);
    });

    test('debe de mostrar correctamente el componente', () => {
        expect( wrapper ).toMatchSnapshot();
        
    })
    
    test('debe de tener un parrafo con el titulo', () => {

        const p = wrapper.find('p');
        console.log(title);
        expect( p.text() ).toBe(title);
    });

    test('debe de tener la imagen y el alt igual al url', () => {

        const img = wrapper.find('img');
        const {src} = img.props();
        console.log(src);
        expect( src ).toBe(url);
    });

    test('debe de tener la clase animate_bounce', () => {
        
        const div = wrapper.find('div');
        const {className} = div.props();
        expect( className.includes('animate__bounce')).toBe(true);
    });
})
