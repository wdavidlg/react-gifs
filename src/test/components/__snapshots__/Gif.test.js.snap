// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <Gif /> debe de mostrar correctamente el componente 1`] = `
<div
  className="gif animate__animated animate__bounce"
>
  <img
    alt="#"
    src="https://localhost/imagen.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
